File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis.VisualBasic.Emit/GenericNamespaceTypeInstanceReference.cs
using Microsoft.Cci;
using Microsoft.CodeAnalysis.VisualBasic.Symbols;

namespace Microsoft.CodeAnalysis.VisualBasic.Emit
{
	internal sealed class GenericNamespaceTypeInstanceReference : GenericTypeInstanceReference
	{
		public override IGenericTypeInstanceReference AsGenericTypeInstanceReference => this;

		public override INamespaceTypeReference AsNamespaceTypeReference => null;

		public override INestedTypeReference AsNestedTypeReference => null;

		public override ISpecializedNestedTypeReference AsSpecializedNestedTypeReference => null;

		public GenericNamespaceTypeInstanceReference(NamedTypeSymbol underlyingNamedType)
			: base(underlyingNamedType)
		{
		}
	}
}

File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis/VisualBasic/Emit/GenericNamespaceTypeInstanceReference.vb
Imports Microsoft.Cci
Imports Microsoft.CodeAnalysis.VisualBasic.Symbols
Imports System

Namespace Microsoft.CodeAnalysis.VisualBasic.Emit
	Friend NotInheritable Class GenericNamespaceTypeInstanceReference
		Inherits GenericTypeInstanceReference
		Public Overrides ReadOnly Property AsGenericTypeInstanceReference As IGenericTypeInstanceReference
			Get
				Return Me
			End Get
		End Property

		Public Overrides ReadOnly Property AsNamespaceTypeReference As INamespaceTypeReference
			Get
				Return Nothing
			End Get
		End Property

		Public Overrides ReadOnly Property AsNestedTypeReference As INestedTypeReference
			Get
				Return Nothing
			End Get
		End Property

		Public Overrides ReadOnly Property AsSpecializedNestedTypeReference As ISpecializedNestedTypeReference
			Get
				Return Nothing
			End Get
		End Property

		Public Sub New(ByVal underlyingNamedType As NamedTypeSymbol)
			MyBase.New(underlyingNamedType)
		End Sub
	End Class
End Namespace